packages/aws-cdk-lib/aws-codebuild/lib/project.ts

```diff
@@ -616,9 +616,10 @@ export interface CommonProjectProps {
   /**
    * Where to place the network interfaces within the VPC.
    *
-   * Only used if 'vpc' is supplied.
+   * Only used if 'vpc' is supplied. If you have no PRIVATE_WITH_EGRESS subnets in the specified `vpc`, leaving this
+   * undefined will require VPC endpoints to access AWS services.
    *
-   * @default - All private subnets.
+   * @default - private subnets if any else public subnets
```

Comment:
this doesn't make sense to me. if any what? if any subnets exist?

Reply:
How about "private subnets if available else public subnets"